Feature/support interactive transitioning

andykkt opened the pull request:
Support interactive transitioning using TranslationAnimationController
- Changed 'duration' to 'animationDuration' because UIPercentDrivenInteractiveTransition has same variable name
- Added interactiveController delegates

andykkt committed:
Merge pull request #5 from BareFeetWare/feature/before_framework

Feature/before framework

Commit:
Support interactive transitioning using TranslationAnimationController
- Changed 'duration' to 'animationDuration' because UIPercentDrivenInteractiveTransition has same variable name
- Added interactiveController delegates

barefeettom commented on BFWControls/Modules/Transition/View/TranslationAnimationController.swift:
```diff
@@ -31,12 +31,13 @@ enum Direction: Int {
 
     @IBInspectable var isPresenting: Bool = true
-    @IBInspectable var duration: TimeInterval = 0.3
+	@IBInspectable var animateDuration: TimeInterval = 0.3
```
As discussed, please rename this `transitionDuration`. I think IBInspectable doesn't like `TimeInterval` (I think it used to, but maybe not since Swift), so please check and if necessary change to `CGFloat`.

andykkt replied:
Fixed in latest commit
- Changed name to transitionDuration and type to CGFloat

barefeettom commented on BFWControls/Modules/Transition/View/TranslationAnimationController.swift:
```diff
@@ -160,7 +162,24 @@ class TranslationAnimationController: NSObject, UIViewControllerAnimatedTransiti
-    
+	
+	func animationController(forPresented presented: UIViewController, presenting: UIViewController, source: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+		self.isPresenting = true
```
Can we delete `self.` here and below?

andykkt replied:
Removed in latest commit

Commit:
Changed animateDuration to transitionDuration and changed type to CGFloat
Removed self

Commit:
Adding examples for simple transition using transitioningDelegateOutlet from storyboard and interactive transitioning

andykkt commented:
Also added examples for transition

Commit:
Added blur background option for transition animation

andykkt commented:
@barefeettom Please review again for latest commit

barefeettom commented on BFWControls.xcodeproj/project.pbxproj:
```diff
@@ -8,6 +8,8 @@
 
 /* Begin PBXBuildFile section */
 		9EAE74CF1EA7053C00349520 /* UIImage+BFW.swift in Sources */ = {isa = PBXBuildFile; fileRef = 9EAE74CE1EA7053C00349520 /* UIImage+BFW.swift */; };
+		A10D2D441EC57E7B006FE9D4 /* transition.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = A10D2D431EC57E7B006FE9D4 /* transition.storyboard */; };
```
Please use an uppercase "T" for the storyboard file name, like the other files.

andykkt replied:
Fixed in latest commit

barefeettom commented on BFWControls/Sample/Transition/TransitionViewController.swift:
```diff
@@ -0,0 +1,77 @@
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
```
Please delete unused functions.

andykkt replied:
Deleted in latest commit

barefeettom commented on BFWControls/Sample/Transition/TransitionViewController.swift:
```diff
@@ -0,0 +1,77 @@
+		case .began:
+			interactiveTransition.isInteractive = true
+			interactiveTransition.direction = .down
+			performSegue(withIdentifier: SegueIdentifier.interactiveSegue.rawValue, sender: self)
```
Please use `sender: pan.view` or `sender: pan`, not `sender: self`

andykkt replied:
Using pan.view for sender

barefeettom commented on BFWControls/Sample/Transition/TransitionViewController.swift:
```diff
@@ -0,0 +1,77 @@
+
+	@IBAction func handlePanGesture(_ pan: UIPanGestureRecognizer) {
+		let translation = pan.translation(in: pan.view!)
+		let d =  translation.y / pan.view!.bounds.height
```
Please use a more descriptive variable name than `d`, eg `changedHeight`

andykkt replied:
Changed to progress, 'd' was short term for 'distance' however 'progress' is more suitable in this context.

barefeettom commented on BFWControls/Sample/Transition/TransitionViewController.swift:
```diff
@@ -0,0 +1,77 @@
+		case .began:
+			interactiveTransition.isInteractive = true
+			interactiveTransition.direction = .down
+			performSegue(withIdentifier: SegueIdentifier.interactiveSegue.rawValue, sender: self)
```
If you conform this view controller to `SegueIdentifierType`, `performSegue` can use the enum `.interactiveSegue` directly instead of `SegueIdentifier.interactiveSegue.rawValue`

andykkt replied:
Using SegueHandler to fix the issue from latest commit

barefeettom commented on BFWControls/Sample/Transition/TransitionViewController.swift:
```diff
@@ -0,0 +1,77 @@
+		}
+    }
+
+
```
Please use a single blank line between functions.

andykkt replied:
Line removed

barefeettom commented on BFWControls/Sample/Transition/TransitionViewController.swift:
```diff
@@ -0,0 +1,77 @@
+
+    // In a storyboard-based application, you will often want to do a little preparation before navigation
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+		guard let segueIdentifier = SegueIdentifier(rawValue: segue.identifier!)
```
Please allow for the identifier to be nil, without crashing. eg:

```swift
guard let identifier = segue.identifier,
    let segueIdentifier = SegueIdentifier(rawValue: identifier)
```

andykkt replied:
Fixed in latest commit

Commit:
Changed transition.storyboard to Transition.storyboard
Deleted unused functions
Used guard let for segue.identifier
Removed redundant function and blank line
Changed variable d to progress
Changed sender to pan.view
Use SegueHandler

andykkt commented:
Please re-review the PR
